src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -556,6 +584,10 @@ export class HttpConnection implements IConnection {
             negotiateUrl += index === -1 ? "?" : "&";
             negotiateUrl += "negotiateVersion=" + this._negotiateVersion;
         }
+        if (negotiateUrl.indexOf("useAck") === -1 && this._options.useAcks === true) {
```

Comment:
Curious, can this whole function be refactored to use `URL` and `URLSearchParams` instead?

Reply:
Refactored since the old code had a bug with adding `?` vs `&`.